lib/matplotlib/axes/_axes.py

```diff
@@ -5376,6 +5433,11 @@ def get_interp_point(ind):
         collection = mcoll.PolyCollection(polys, **kwargs)
 
         # now update the datalim and autoscale
+        # For between pad last value
+        if step == 'between':
+            x1 = np.append(x1, x1[-1])
+            x2 = np.append(x2, x2[-1])
+            where = np.append(where, True)
```

Comment:
wrt repeated code, almost all the changes you made here were also made, verbatim as far as I can tell, in `ax.step`.   To me, that says there should ideally be a helper for both methods.

Reply:
I don't understand what you mean. What you highlight, is _kind of_ duplicate between `fill_between` and `fill_betweenx` which are well _kind of_ duplicates, but probably doesn't make sense trying to merge them. 

There's almost no code added in `step` itself. Really only this:

```
        cbook._check_in_list(('pre', 'post', 'mid', 'between', 'edges'),
                             where=where)

        if where in ['between', 'edges']:
            if len(x) == len(y) or abs(len(x)-len(y)) > 1:
                raise ValueError(f"When plotting with 'between' or 'edges'"
                                 f"input sizes have to be have to satisfy "
                                 f"len(x) + 1 == len(y) or "
                                 f"len(x) == len(y) + 1 but x "
                                 f"and y have size {len(x)} and {len(y)}")
```

Reply:
I'm sorry you are correct, the code repeat is between the x/y versions of `fill_between` and `fill_betweenx`.  Its getting complicated enough I suspect it *is* worth merging them.

Reply:
While I think it would be possible, the reduction in duplication would be paid in disproportionate increase in complexity, which is imo not-desirable. Not to mention merging `fill_between` and `fill_betweenx` would be considerable API change

Reply:
I'm suggesting something like `polys=_get_fill_between_polys(x1, x2, y, dir='y')`, where for `fill_bwteenx` you just need to call with `y1, y2, x, dir='x` and transpose the x and y in the polys before returning them for the "other" implementation.

Reply:
Thanks, I think this helps.  Only concern is that I'm not sure helpers that are specific to methods should go in `cbook`.  Probably should just go near `fill_between`?

Reply:
That's ofc possible, but you do have dedicated helpers in cbook already for violin/box plots and quite possibly many are not marked as such

Reply:
maybe we need an `axes/_helpers.py` module that we could put these in.  Once upon a time `cbook` was supposed to be for cook-book routines that actually users would use...  Some other dev input would be helpful here.

Reply:
Yeah, we will move things like this into `axes/_util.py`, but no need to do that here unless you really want to.  

Speaking of which, I think the other helper functions you've added to `cbook` need to be private so we don't need to worry about the API.

Reply:
I'm ok with that, could even look into it, but I don't think this is necessarily related to the subject of this PR

Reply:
Sorry I meant your new other helper functions you are adding to cbook.

Reply:
I don't understand what the change you are asking for is? The `pad_arrays` is not necessarily method specific I think.

Reply:
It should be private.

Reply:
Hmm, alright, but shouldn't it all be fixed at once? There are enough changes as is in this PR.